operators/src/processing/temporal_raster_aggregation.rs

```diff
@@ -141,85 +160,113 @@ where
 {
     type RasterType = P;
 
+    #[allow(clippy::too_many_lines)]
     fn raster_query<'a>(
         &'a self,
         query: crate::engine::QueryRectangle,
         ctx: &'a dyn crate::engine::QueryContext,
     ) -> Result<futures::stream::BoxStream<'a, Result<RasterTile2D<Self::RasterType>>>> {
         match self.aggregation_type {
-            AggregationType::Min => {
-                if self.ignore_no_data {
-                    let spec = TemporalRasterAggregationSubQuery {
-                        fold_fn: no_data_ignoring_fold_future::<P, MinIgnoreNoDataAccFunction>,
-                        no_data_value: self.no_data_value,
-                        initial_value: P::max_value(),
-                        step: self.window,
-                    };
-
-                    let s = RasterOverlapAdapter::<'a, P, _, _>::new(
+            Aggregation::Min {
+                ignore_no_data: true,
+            } => Ok(self
+                .create_subquery(
+                    no_data_ignoring_fold_future::<P, MinIgnoreNoDataAccFunction>,
+                    P::max_value(),
+                )
+                .into_raster_overlap_adapter(&self.source, query, ctx, self.tiling_specification)
+                .boxed()),
+            Aggregation::Min {
+                ignore_no_data: false,
+            } => Ok(self
+                .create_subquery(fold_future::<P, MinAccFunction>, P::max_value())
+                .into_raster_overlap_adapter(&self.source, query, ctx, self.tiling_specification)
+                .boxed()),
+            Aggregation::Max {
+                ignore_no_data: true,
+            } => Ok(self
+                .create_subquery(
+                    no_data_ignoring_fold_future::<P, MaxIgnoreNoDataAccFunction>,
+                    P::min_value(),
+                )
+                .into_raster_overlap_adapter(&self.source, query, ctx, self.tiling_specification)
+                .boxed()),
+            Aggregation::Max {
+                ignore_no_data: false,
+            } => Ok(self
+                .create_subquery(fold_future::<P, MaxAccFunction>, P::min_value())
+                .into_raster_overlap_adapter(&self.source, query, ctx, self.tiling_specification)
+                .boxed()),
+            Aggregation::First {
+                ignore_no_data: true,
```

Comment:
shouldn't it be the other way around? Ignoring no data mean just taking the pixels from the first tile no matter if they are no data (ignoring no data).

Reply:
ok, you designed the ignore_no_data semantics

Reply:
I am still unsure. Take this example:
- NDVI data set starts at 1.1.2014
- set query time 1.12.2013: result is all no_data
- add temporal aggregation operator, window 2 month, type: First, ignore_no_data: false
- current result: we get the image from 1.1.2014

however, ignore_no_data ist only another way of saying: take only valid data. If it is false, then we also take no_data values into the aggregation. Thus, in this case we should take the first raster values, regardless if there are no_data. Right?

That's actually also what I wanted in my comment above, but it seems like it was right before.